crates/matrix-sdk/tests/integration/widget.rs

```diff
@@ -679,8 +673,67 @@ async fn test_send_delayed_state_event() {
     assert_eq!(msg["action"], "send_event");
     let delay_id = msg["response"]["delay_id"].as_str().unwrap();
     assert_eq!(delay_id, "1234");
+}
+
+#[async_test]
+async fn test_fail_sending_delay_rate_limit() {
+    let (_, mock_server, driver_handle) = run_test_driver(false).await;
+
+    negotiate_capabilities(
+        &driver_handle,
+        json!([
+            "org.matrix.msc4157.send.delayed_event",
+            "org.matrix.msc2762.send.event:m.room.message"
+        ]),
+    )
+    .await;
+
+    Mock::given(method("PUT"))
+        .and(path_regex(r"^/_matrix/client/v3/rooms/.*/send/m.room.message/.*$"))
```

Comment:
We have a mock in the `MatrixMockServer` called `.mock_room_send()`, can you use that instead?

Reply:
This has been resolved without explanation and without a change to the code, can you comment at least please? :)

Reply:
I fixed this locally clicked resolve and then noticed I also want to create `with_delay` mocks to replace a couple of more tests with the new mocking pattern.